src/breaker/CountBreaker.ts

```diff
@@ -0,0 +1,124 @@
+import { CircuitState } from '../CircuitBreakerPolicy';
+import { IBreaker } from './Breaker';
+
+export interface ICountBreakerOptions {
+  /**
+   * Percentage (from 0 to 1) of requests that need to fail before we'll
+   * open the circuit.
+   */
+  threshold: number;
+
+  /**
+   * Size of the count based sliding window.
+   */
+  size: number;
+
+  /**
+   * Minimum number of calls needed to (potentially) open the circuit.
+   * Useful to avoid unnecessarily tripping when there are only few samples yet.
+   * Defaults to {@link ICountBreakerOptions.size}.
+   */
+  minimumNumberOfCalls?: number;
+}
+
+export class CountBreaker implements IBreaker {
+  private readonly threshold: number;
+  private readonly minimumNumberOfCalls: number;
+
+  /**
+   * The samples in the sliding window. `true` means "success", `false` means
+   * "failure" and `undefined` means that there is no sample yet.
+   */
+  private readonly samples: (boolean | undefined)[];
+  private successes = 0;
+  private failures = 0;
+  private currentSample = 0;
+
+  /**
+   * CountBreaker breaks if more than `threshold` percentage of the last `size`
+   * calls failed, so long as at least `minimumNumberOfCalls` calls have been
+   * performed (to avoid opening unnecessarily if there are only few samples
+   * in the sliding window yet).
+   */
+  constructor({ threshold, size, minimumNumberOfCalls = size }: ICountBreakerOptions) {
+    if (threshold <= 0 || threshold >= 1) {
+      throw new RangeError(`CountBreaker threshold should be between (0, 1), got ${threshold}`);
```

Comment:
```suggestion
      throw new RangeError(`CountBreaker threshold should be in the range [0, 1], got ${threshold}`);
```

Reply:
The range is actually exclusive, so imo, it makes sense to keep the parenthesis instead of brackets (since that's pretty common notation for open intervals).

On the other hand, there is an argument to be made for making the range inclusive (`threshold < 0 || threshold > 1)`. It is however also [exclusive in the SamplingBreaker](https://github.com/connor4312/cockatiel/blob/master/src/breaker/SamplingBreaker.ts#L46) (I basically just copied it from there).

What do you think?

Reply:
Oh, nevermind, I misread.